### PR TITLE
map tile request cancellation

### PR DESCRIPTION
**Pull request by cmcnicholas**
allows cancellation of ongoing tile requests if the user pans the extent outside of the requested tiles bounds or on zoom.

removed `Svr` because we don't need it, just the fetch param factories.

### CODE REVIEW

**Comment by cmcnicholas** [comment]
Also fixes AL-3166

**Comment by OlegRayskiy** [comment]
> Also fixes AL-3166

But does it fix tests?

File: src/map/layers/cache/AlloyTileRequestCache.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  }
+
+  /**
+   * clears an entire zoom level of cache and cancels all requests on that zoom level
+   * @param zoom the zoom level to clear
+   */
+  public clearZoom(zoom: number): void {
+    const requestsAtZoom = this.requests.get(zoom);
+    if (requestsAtZoom) {
+      [...requestsAtZoom.values()].forEach((request) => request.cancel());
```
Comment (OlegRayskiy) [inline src/map/layers/cache/AlloyTileRequestCache.ts]: Use Array.from

File: src/map/layers/cache/AlloyTileRequestCache.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      this.requests.delete(zoom);
+    }
+  }
+
+  /**
+   * clears all zoom levels outside of a given zoom and cancels all requests in those levels
+   * @param zoom the zoom level to maintain the cache for
+   */
+  public clearOutsideZoom(zoom: number): void {
+    [...this.requests.keys()].filter((key) => key !== zoom).forEach((key) => this.clearZoom(key));
```
Comment (OlegRayskiy) [inline src/map/layers/cache/AlloyTileRequestCache.ts]: Array.from

File: src/map/layers/cache/AlloyTileRequestCache.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    this.clearOutsideZoom(zoom);
+
+    // now get the zoom level of the tiles
+    const requestsAtZoom = this.requests.get(zoom);
+    if (requestsAtZoom) {
+      // generate tile keys
+      const tileKeysToKeep = new Set(tiles.map((tile) => tile.requestKey));
+
+      // cancel and remove any requests outside the tiles
+      [...requestsAtZoom.values()]
```
Comment (OlegRayskiy) [inline src/map/layers/cache/AlloyTileRequestCache.ts]: Array.from

Comment (cmcnicholas) [reply src/map/layers/cache/AlloyTileRequestCache.ts]: coo, nice catch

**Pull request merged**